shared/fs/common/filename-text.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+import {namedConnect} from '../../util/container'
+import * as Types from '../../constants/types/fs'
+import * as Constants from '../../constants/fs'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
+import * as React from 'react'
+import {allTextTypes} from '../../common-adapters/text.shared'
+
+type TextType = $Keys<typeof allTextTypes>
+
+type OwnProps = {|
+  path?: Types.Path,
+  filename?: string,
+  selectable?: boolean,
+  style?: Styles.StylesCrossPlatform,
+  type: TextType,
+|}
+
+type Props = {|
+  fileExtension: string,
+  fileNameWithoutExtension: string,
+  selectable?: boolean,
+  style?: Styles.StylesCrossPlatform,
+  type: TextType,
+|}
+
+const FilenameText = (props: Props) => (
+  <Kb.Box2 direction="horizontal" fullWidth={true}>
+    <Kb.Text
+      type={props.type}
+      style={Styles.collapseStyles([props.style, styles.breakAll])}
+      lineClamp={1}
+      selectable={props.selectable}
+    >
+      {props.fileNameWithoutExtension}
+    </Kb.Text>
+    {props.fileExtension && (
+      <Kb.Text type={props.type} style={props.style} selectable={props.selectable} lineClamp={0}>
+        {props.fileExtension}
+      </Kb.Text>
+    )}
+  </Kb.Box2>
+)
+
+const mapStateToProps = state => ({})
+
+const mapDispatchToProps = dispatch => ({})
+
+const mergeProps = (s, d, o) => {
+  const [fileNameWithoutExtension, fileExtension] = Constants.splitFileNameAndExtension(
+    o.path ? Types.getPathName(o.path) : o.filename || ''
+  )
+  return {
+    fileExtension,
+    fileNameWithoutExtension,
+    selectable: o.selectable,
+    style: o.style,
+    type: o.type,
+  }
+}
```

Comment:
Here you don't need state from the redux store, or dispatch any actions. So there's no need to have a redux container. You can just do this inside the component. It saves Redux from iterating over this container each time a state update happens, and saves us from having to have those story providers for this.